packages/trpc/tests/deleteCredential.test.ts

```diff
@@ -0,0 +1,164 @@
+import MockDataBaseClient from "../../../tests/libs/__mocks__/mockDatabaseClient";
+
+import { addUsers, addEventTypesToDb } from "@calcom/web/test/utils/bookingScenario/bookingScenario";
+
+import { describe, test, expect } from "vitest";
+
+import { createContextInner } from "../server/createContext";
+import { createCaller } from "../server/routers/_app";
+
+const testUser = {
+  id: 1,
+  email: "test@test.com",
+  username: "test-user",
+};
+
+const setupIndividualCredentialTest = async () => {
+  const ctx = await createContextInner({
+    locale: "en",
+    session: {
+      hasValidLicense: true,
+      upId: "test-upId",
+      user: {
+        id: testUser.id,
+        profile: {
+          id: 1,
+          upId: "profile-upid",
+          username: testUser.username,
+        },
+      },
+    },
+  });
+
+  const caller = createCaller(ctx);
```

Comment:
Using tRPC inside of unit tests feels off to me. Would prefer we move towards "thin controllers", in this case "thin handlers", that call off to core application services that handle this logic. tRPC is just the technological mechanism to making an HTTP request so having it in our unit tests that's purpose are to test core logic, not tRPC, is not ideal.

I don't think it's too much of a lift to move basically everything in the handler to a different class/file that is an application service.

Reply:
i.e. We should have a suite of tests that can test our application logic regardless of technology. If we move that tRPC handler to RSC, for instance, now your test breaks and it shouldn't